pkg/operator/ceohelpers/machine_api.go

```diff
@@ -0,0 +1,41 @@
+package ceohelpers
+
+import (
+	machinelistersv1beta1 "github.com/openshift/client-go/machine/listers/machine/v1beta1"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/utils/pointer"
+)
+
+// IsMachineAPIFunctionalWrapper returns a function for determining if the Machine API is functional.
+// As of today Machine API is functional when we find Machines resources in the Running state.
+//
+// Note:
+//  since this function needs to take into account only master machine objects
+//  make sure machineInformer contain only filtered data
+//  otherwise it might be expensive to react to every node update in larger installations
+func IsMachineAPIFunctionalWrapper(masterMachineInformer cache.SharedInformer, masterMachineLister machinelistersv1beta1.MachineLister) func() (bool, error) {
```

Comment:
@JoelSpeed could you please take a look at this function and check if it makes sense?